src/Antiforgery/test/DefaultAntiforgeryTest.cs

```diff
@@ -348,6 +348,41 @@ public void GetAndStoreTokens_ExistingCachingHeaders_Overriden()
             Assert.Equal("no-cache", context.HttpContext.Response.Headers[HeaderNames.Pragma]);
         }
 
+        [Theory]
+        [InlineData("Cache-Control", "no-cache, no-store, must-revalidate, private")]
+        [InlineData("Pragma", "no-cache")]
+        public void GetAndStoreTokens_ExistingNoCachingHeaders_NotOverriden(string headerName, string headerValue)
```

Comment:
Could you split this as follows: 


* `GetAndStoreTokens_DoesNotOverwriteCacheControlHeader`
* `GetAndStoreTokens_OverwritesCacheControlHeader_IfNoStoreIsNotSet`
* `GetAndStoreTokens_OverwritesCacheControlHeader_IfNoCacheIsNotSet`
* `GetAndStoreTokens_DoesNotOverwritePragmaHeader`

You could use a private method to share all of the boiler plate in `Arrange & Act`.